qCC/mainwindow.cpp

```diff
@@ -839,6 +839,7 @@ void MainWindow::doActionColorFromScalars()
 			ccColorFromScalarDlg* cfsDlg = new ccColorFromScalarDlg(this, cloud);
 			cfsDlg->setAttribute(Qt::WA_DeleteOnClose, true);
 			cfsDlg->show();
+			cfsDlg->setupDisplay(); //required otherwise the initial histograms are grey
```

Comment:
Kind of nitpicky, but it feels like this is in the wrong place. If the dialog doesn't work properly without this being called, then it should be called somewhere in the dialog class.

Does it work if `setupDisplay` is called at the end of the constructor?

If not, we should override show() and call it there the first time the dialog is shown.

Now - we only use this dialog here, so if you think it's overkill to fix properly I don't feel too strongly - about a 5.75 on this one. :-)

(In fact that `setAttribute` should be done in the constructor as well.)

Reply:
Easy fix. I had fixed the original issue that formerly required this. I was able to just remove it.

I also moved the setAttribute to the constructor